docs/apis-hook.md

```diff
@@ -4,4 +4,149 @@ sidebar_position: 4
 
 # APIs hook
 
-This page will discuss - How to interact with the ReactPlay data using custom hook.
+In React Play, we work with APIs by using our custom-made react hooks. They are as follows:
+
+1. [useCacheResponse](#usecacheresponse)
+1. [useContributors](#usecontributors)
+1. [useFeaturedPlays](#usefeaturedplays)
+1. [useFetch](#usefetch)
+1. [useGetPlays](#usegetplays)
+1. [useGitHub](#usegithub)
+1. [useLikePlays](#uselikeplays)
+1. [useLocalStorage](#uselocalstorage)
+
+In this section, we will discuss each of them in detail.
+
+## useCacheResponse
+
+As the name suggests, this hook is used to create and maintain a cached version of the data.
+
+```JavaScript
+  const [getCacheResponse, setCacheResponse] = useCacheResponse();
+  const response = axios(API_URL);
+  setCacheResponse(FILTER_DATA_RESPONSE, response);
+  // ....
+ const isCachedResponse = getCacheResponse(FILTER_DATA_RESPONSE);
+```
+
+`useCachedResponse()` returns a getter and a setter function. In the above example, we are setting the `FILTER_DATA_RESPONSE` value based on the API response. This value is now stored as a cached response. We can then use the getter function to access this cached value by passing the same key of `FILTER_DATA_RESPONSE`
+
+## useContributors
+
+This custom hook helps in retrieving the names of the contributors in [react play](https://github.com/reactplay/react-play) repository on GitHub. `useContributors` takes one parameter `sorted` which is a boolean value. If `true` is passed, the result is sorted in the ascending order of the number of contributions by the contributor. This function returns an object with `{ data, error, isLoading }`.
+
+```JavaScript
+ const { data, error, isLoading } = useContributors(true);
+
+return (
+  <ul className="list-contributors">
+  {isLoading && <li>Loading...</li>}
+  {error && <li>Error: {error.message}</li>}
+  {data &&
+    data.map((contributor) => (
+      <li
+        className="contributor"
+        data-testid={`contributor-${contributor.id}`}
+        key={contributor.id}
+      >
+      {contributor.login}
+      </li>
+   ))}
+</ul>
+)
+```
+
+## useFeaturedPlays
+
+Invoking `useFeaturedPlays` hook would internally run a GraphQL query to retrieve the list of featured plays. This function returns an array of 3 items. A `loading` state is used to represent a loader while the data is being fetched. The `error` object contains details about any error that occurred while trying to fetch the data. The `data` object contains the list of featured plays as an array.
```

Comment:
I think for every APIs, providing the parameter details as bulleted points will be the better way.

- A `loading` state represents a loader while the data is fetched.
- The `error` object contains details about any error that occurred while trying to fetch the data.
- The `data` object contains the list of featured plays as an array.